### ISSUE TITLE
EFS doesn't pass tags in creation request 

### ISSUE THREAD

**Issue by splintercat** [terraform-providers/terraform-provider-aws#7662]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

Terraform v0.11.11
+ provider.aws v1.60.0

### Affected Resource(s)

* aws_efs_file_system

### Terraform Configuration Files

```hcl

provider "aws" {
  region = "us-west-2"
}

resource "aws_efs_file_system" "efs_example" {
  tags = {
    Name = "catscratch-efs-system"
  }
}


```

### Debug Output

`TF_LOG=trace` output
https://gist.github.com/splintercat/e8dc785140320438754e97be7ef3fd53

### Expected Behavior

Terraform creates an EFS filesystem with the tags as part of the creation request.

### Actual Behavior

Terraform apparently creates a resource and then tags it.  Which fails with our policy since we are depending on the tags being passed in with the creation request. 

The policy is below but the basics is that to create an EFS drive you have to pass a tag with certain kinds of values in as part of the resource request.

### Steps to Reproduce

1. Have policy that's in the Important Factoids below attached to your role or user
1. `terraform apply`

### Important Factoids

Our company is using tags to control what teams can create.  In this case for auditing purposes they have to tag their resource with a Name tag that starts with their team name.

#### Relevant AWS Policy
```json
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Sid": "efsSplatResource",
            "Effect": "Allow",
            "Action": "elasticfilesystem:Describe*",
            "Resource": "*"
        },
        {
            "Sid": "createEFS",
            "Effect": "Allow",
            "Action": "elasticfilesystem:CreateFileSystem",
            "Resource": "*",
            "Condition": {
                "StringLike": {
                    "aws:RequestTag/Name": "catscratch-*"
                }
            }
        },
        {
            "Sid": "efsWithNameTag",
            "Effect": "Allow",
            "Action": [
                "elasticfilesystem:UpdateFileSystem",
                "elasticfilesystem:PutLifecycleConfiguration",
                "elasticfilesystem:ModifyMountTargetSecurityGroups",
                "elasticfilesystem:DescribeMountTargets",
                "elasticfilesystem:DescribeMountTargetSecurityGroups",
                "elasticfilesystem:DescribeLifecycleConfiguration",
                "elasticfilesystem:DeleteTags",
                "elasticfilesystem:DeleteMountTarget",
                "elasticfilesystem:DeleteFileSystem",
                "elasticfilesystem:CreateTags",
                "elasticfilesystem:CreateMountTarget"
            ],
            "Resource": "*",
            "Condition": {
                "StringLike": {
                    "aws:ResourceTag/Name": "catscratch-*"
                }
            }
        },
        {
            "Sid": "efsRequiredEC2NetworkStuff",
            "Effect": "Allow",
            "Action": [
                "ec2:DescribeSubnets",
                "ec2:DescribeNetworkInterfaces",
                "ec2:DeleteNetworkInterface",
                "ec2:CreateNetworkInterface"
            ],
            "Resource": "*"
        }
    ]
}
```

Creating the EFS drive will work just fine with the AWS Cli (version 1.16.111)
```

$ aws efs create-file-system --tags "Key=Name,Value=catscratch-efs-system" --creation-token 
"sometoken"

{
    "SizeInBytes": {
        "ValueInIA": 0, 
        "ValueInStandard": 0, 
        "Value": 0
    }, 
    "Name": "catscratch-efs-system", 
    "CreationToken": "sometoken", 
    "Encrypted": false, 
    "Tags": [
        {
            "Value": "catscratch-efs-system", 
            "Key": "Name"
        }
    ], 
    "CreationTime": 1550889875.0, 
    "PerformanceMode": "generalPurpose", 
    "FileSystemId": "fs-ea90110a", 
    "NumberOfMountTargets": 0, 
    "LifeCycleState": "creating", 
    "OwnerId": "672714205403", 
    "ThroughputMode": "bursting"
}

```

### References
Docs for EFS policy
https://docs.aws.amazon.com/IAM/latest/UserGuide/list_amazonelasticfilesystem.html

### Similar issues:
EBS Volume Creation does nopt honor aws:RequestTag : #6179 

* #0000

**Comment by ewbankkit** [comment]
It looks like EFS tagging during creation was added in [AWS SDK v1.17.1](https://github.com/aws/aws-sdk-go/releases/tag/v1.17.1).

**Comment by ewbankkit** [comment]
Announcement [here](https://aws.amazon.com/about-aws/whats-new/2019/02/amazon-efs-now-supports-tag-on-create/).

**Comment by ewbankkit** [comment]
@splintercat It should "just" be a case of adding something like
```go
Tags: tagsFromMapEFS(d.Get("tags").(map[string]interface{})),
```
to [`createOpts`](https://github.com/terraform-providers/terraform-provider-aws/blob/aa7fc99ccef3608264d423ba1c3ee86c656315a3/aws/resource_aws_efs_file_system.go#L132-L135) in [`resourceAwsEfsFileSystemCreate()`](https://github.com/terraform-providers/terraform-provider-aws/blob/aa7fc99ccef3608264d423ba1c3ee86c656315a3/aws/resource_aws_efs_file_system.go#L121) and removing the call to [`setTagsEFS()`](https://github.com/terraform-providers/terraform-provider-aws/blob/aa7fc99ccef3608264d423ba1c3ee86c656315a3/aws/resource_aws_efs_file_system.go#L196-L199).

**Comment by bflad** [comment]
Support for `aws_efs_file_system` resource tagging on creation has been merged and will release with version 2.31.0 of the Terraform AWS Provider, later next week. Thanks to @mildwonkey for the implementation.
